libunwind/src/Registers.hpp

```diff
@@ -4544,6 +4607,14 @@ class _LIBUNWIND_HIDDEN Registers_riscv {
   void        setSP(reg_t value) { _registers[2] = value; }
   reg_t       getIP() const { return _registers[0]; }
   void        setIP(reg_t value) { _registers[0] = value; }
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(_LIBUNWIND_SANDBOX_OTYPES)
+  reg_t       getUnsealedECSP(uintptr_t sealer) { return getSP(); }
```

Comment:
This makes no sense. CHERI-RISC-V has no such thing as an executive mode, so it should not have methods involving an executive stack pointer.

Reply:
Yeah it was shim to make it build on CHERI-RISC-V. It's called `ECSP` because it was originally made for Morello, but would probably work similarly in CHERI-RISC-V (assuming otypes end up in there and we make use of them in libunwind). I'll rename usage of `ECSP` to `TrustedStack` or something along those lines to relay what it's supposed to mean. Alternatively, if we really don't want to make use of otypes I could just gate all calls behind a Morello `ifdef`.